File: vera-shawn/tipCalculator/tipCalculator.js
# Create tip calculator #

**Procedural**

START

INPUT user input of bill amount

CONVERT calculate % amount of given INPUT

OUTPUT float number in dollar amount

OUTPUT message to user if NaN

END

**Functional**

*Variables:*

tip

terribleTip

poorTip

goodTip

greatTip

excellentTip

*Functions*

```javaScript function terribleTip(amount) {
    if (tip <= terribleTip) {
        return "Tip more please"
    }
} ```


function calculateTip(amount, rating) {
    //each statement returns the tip amount based on the service
    let rate = rating.toLowerCase();
    if (rate === 'excellent') {
      return Math.ceil(amount*20/100);
    }
    else if (rate === 'great') {
      return Math.ceil(amount*15/100);
    }
    else if (rate === 'good') {
      return Math.ceil(amount*10/100);
    }
    else if (rate === 'poor') {
      return Math.ceil(amount*5/100);
    }
    else if (rate === 'terrible') {
      return Math.ceil(amount*0/100);
    }
    else {
      return ("Rating not recognised");
    }
  }

File: vera-shawn/tip-calculator.py
# begin
# convert javascript solution to python
# end

import math


def calculate_tip(amount, rating):
    # each statement returns the tip amount based on the service
    rate = rating.lower()
    if rate == "excellent":
      return math.ceil(amount*20/100)
    elif rate == 'great':
      return math.ceil(amount*15/100)
    elif rate == 'good':
      return math.ceil(amount*10/100)
    elif rate == 'poor':
      return math.ceil(amount*5/100)
    elif rate == 'terrible':
      return math.ceil(amount*0/100)
    else:
      return ("Rating not recognised")